lib/mongo/error/connection_not_exist.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+# Copyright (C) 2020 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Error
+
+    # Exception raised when trying to check out connection that does not exist.
+    class ConnectionNotExist < Error
```

Comment:
We have no *NotExist and no *NotExists exceptions, but we have a bunch of Missing* ones.  I think `MissingConnection` might be more in line with other exceptions and avoid the singular/plural awkwardness?

Reply:
Thank you, fixed.